### PR TITLE
Add paused, resumed, and seeked emitters.

### PR DESCRIPTION
**Pull request by karyeet**
Very useful for duration played handling.
Not familiar with TS so corrections are likely needed, as well as additions to the interface section at the bottom.

### CODE REVIEW

File: src/lib/Player.ts [hunk 1]
```diff
@@ -102,6 +102,8 @@ export class Player extends EventEmitter {
     public async pause(pause: boolean): Promise<boolean> {
         const d = await this.send("pause", { pause });
         this.paused = pause;
+        if (this.listenerCount("paused") && pause === true) this.emit("paused");
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
        if (this.listenerCount("pause")) this.emit("pause", pause);
```

File: src/lib/Player.ts [hunk 2]
```diff
@@ -102,6 +102,8 @@ export class Player extends EventEmitter {
...
+        if (this.listenerCount("resumed") && pause === false) this.emit("resumed");
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
```

File: src/lib/Player.ts [hunk 3]
```diff
@@ -127,6 +129,7 @@ export class Player extends EventEmitter {
      * @param position Seeks the song to the position specified in milliseconds, use the duration of the song from lavalink to get the duration
      */
     public seek(position: number): Promise<boolean> {
+        if (this.listenerCount("seeked")) this.emit("seeked", position);
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
        if (this.listenerCount("seek")) this.emit("seek", position);
```

**Review by MrJacz — CHANGES_REQUESTED** [review]
You will also need to add the events to the Interface for `Player`.

**Comment by MrJacz** [comment]
Please also fix the TSC & Typedoc errors. and remove the draft status

**Comment by karyeet** [comment]
Don't know whats causing the tsc errors, unfortunately.

File: src/lib/Player.ts [hunk 4]
```diff
@@ -179,27 +181,35 @@ export interface Player {
     on(event: "event", listener: (data: LavalinkEvent) => void): this;
     on(event: "start", listener: (data: LavalinkEvent) => void): this;
     on(event: "end", listener: (data: LavalinkEvent) => void): this;
+    on(event: "pause", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    on(event: "pause", listener: (pause: boolean) => void): this;
```

File: src/lib/Player.ts [hunk 5]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
+    on(event: "seek", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    on(event: "seek", listener: (position: number) => void): this;
```

File: src/lib/Player.ts [hunk 6]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
     on(event: "error", listener: (error: LavalinkEvent) => void): this;
     on(event: "warn", listener: (warning: string) => void): this;
     on(event: "playerUpdate", listener: (data: { state: LavalinkPlayerState; }) => void): this;
 
     once(event: "event", listener: (data: LavalinkEvent) => void): this;
     once(event: "start", listener: (data: LavalinkEvent) => void): this;
     once(event: "end", listener: (data: LavalinkEvent) => void): this;
+    once(event: "pause", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    once(event: "pause", listener: (pause: boolean) => void): this;
```

File: src/lib/Player.ts [hunk 7]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
+    once(event: "seek", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    once(event: "seek", listener: (position: number) => void): this;
```

File: src/lib/Player.ts [hunk 8]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
     once(event: "error", listener: (error: LavalinkEvent) => void): this;
     once(event: "warn", listener: (warning: string) => void): this;
     once(event: "playerUpdate", listener: (data: { state: LavalinkPlayerState; }) => void): this;
 
     off(event: "event", listener: (data: LavalinkEvent) => void): this;
     off(event: "start", listener: (data: LavalinkEvent) => void): this;
     off(event: "end", listener: (data: LavalinkEvent) => void): this;
+    off(event: "pause", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    off(event: "pause", listener: (pause: boolean) => void): this;
```

File: src/lib/Player.ts [hunk 9]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
+    off(event: "seek", listener: (data: LavalinkEvent) => void): this;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    off(event: "seek", listener: (position: number) => void): this;
```

File: src/lib/Player.ts [hunk 10]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
     off(event: "error", listener: (error: LavalinkEvent) => void): this;
     off(event: "warn", listener: (warning: string) => void): this;
     off(event: "playerUpdate", listener: (data: { state: LavalinkPlayerState; }) => void): this;
 
     emit(event: "event", data: LavalinkEvent): boolean;
     emit(event: "start", data: LavalinkEvent): boolean;
     emit(event: "end", data: LavalinkEvent): boolean;
+    emit(event: "pause", data: LavalinkEvent): boolean;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    emit(event: "pause", pause: boolean): boolean;
```

File: src/lib/Player.ts [hunk 11]
```diff
@@ -179,27 +181,35 @@ export interface Player {
...
+    emit(event: "seek", data: LavalinkEvent): boolean;
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
    emit(event: "seek", position: number): boolean;
```

**Review by MrJacz — CHANGES_REQUESTED** [review]
Commit these suggestions and you should be good to go!

**Review by karyeet — COMMENTED** [review]
I think we're good now.

File: src/lib/Player.ts [hunk 12]
```diff
@@ -127,6 +128,7 @@ export class Player extends EventEmitter {
      * @param position Seeks the song to the position specified in milliseconds, use the duration of the song from lavalink to get the duration
      */
     public seek(position: number): Promise<boolean> {
```
Comment (MrJacz) [inline src/lib/Player.ts]: I think, this is a better idea, so it doesn't emit seek if the send doesn't actually send to lavalink. just to be safe
```suggestion
    public async seek(position: number): Promise<boolean> {
        const d = await this.send("seek", { position });
        if (this.listenerCount("seek")) this.emit("seek", position);
        return d;
    }
```

**Review by MrJacz — CHANGES_REQUESTED** [review]
Sorry one more thing

File: src/lib/Player.ts [hunk 13]
```diff
@@ -126,7 +127,12 @@ export class Player extends EventEmitter {
      * Seeks the current song to a certain position
      * @param position Seeks the song to the position specified in milliseconds, use the duration of the song from lavalink to get the duration
      */
-    public seek(position: number): Promise<boolean> {
+    public async seek(position: number): Promise<boolean> {
+        const d = await this.send("seek", { position });
+        if (this.listenerCount("seek")) this.emit("seek", position);
+        return d;
+    }
+        if (this.listenerCount("seek")) this.emit("seek", position);
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
```

File: src/lib/Player.ts [hunk 14]
```diff
@@ -126,7 +127,12 @@ export class Player extends EventEmitter {
...
         return this.send("seek", { position });
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
```

File: src/lib/Player.ts [hunk 15]
```diff
@@ -126,7 +127,12 @@ export class Player extends EventEmitter {
...
     }
```
Comment (MrJacz) [inline src/lib/Player.ts]:
```suggestion
```

**Review by MrJacz — APPROVED** [review]
LGTM

**Pull request merged**